File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    // T: O(n^2) where n is the number of elements in nums
    // AUX: O(n) where n is the number of elements in nums
    public int[] productExceptSelf(int[] nums) {
        int[] result = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            int product = 1;
            for (int j = 0; j < nums.length; j++) {
                if (j != i) {
                    product *= nums[j];
                }
            }
            result[i] = product;
        }

        return result;
    }
}  

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        seen = set()
        k, i = 0, 0

        while i < len(nums):
            if nums[i] not in seen:
                seen.add(nums[i])
                nums[k] = nums[i]
                k += 1
            i += 1
        
        return k
